src/Resources/ResourceManager/Implementation/DeploymentScripts/RemoveAzDeploymentScript.cs

```diff
@@ -0,0 +1,101 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Components;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels;
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.DeploymentScripts
+{
+    [Cmdlet(VerbsCommon.Remove, AzureRMConstants.AzureRMPrefix + "DeploymentScript", SupportsShouldProcess = true, DefaultParameterSetName = RemoveAzDeploymentScript.RemoveDeploymentScriptByName), OutputType(typeof(PsDeploymentScript))]
+    public class RemoveAzDeploymentScript : DeploymentScriptCmdletBase
+    {
+        #region Cmdlet Parameters and Parameter Set Definitions
+
+        internal const string RemoveDeploymentScriptByName = "RemoveDeploymentScriptLogByName";
+        internal const string RemoveDeploymentScriptByResourceId = "RemoveDeploymentScriptLogByResourceId";
+        internal const string RemoveDeploymentScriptByInputObject = "RemoveDeploymentScriptLogByInputObject";
+
+        [Parameter(Position = 0, ParameterSetName = RemoveDeploymentScriptByName, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Position = 1, ParameterSetName = RemoveDeploymentScriptByName, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The name of the resource group.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Alias("ResourceId")]
+        [Parameter(Position = 0, ParameterSetName = RemoveDeploymentScriptByResourceId, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The fully qualified resource Id of the deployment script. Example: /subscriptions/{subId}/resourceGroups/{rgName}/providers/Microsoft.Resources/deploymentScripts/{deploymentScriptName}")]
+        [ValidateNotNullOrEmpty]
+        public string Id { get; set; }
```

Comment:
```suggestion
        [Alias("Id")]
        [Parameter(Position = 0, ParameterSetName = RemoveDeploymentScriptByResourceId, Mandatory = true, ValueFromPipelineByPropertyName = true,
            HelpMessage = "The fully qualified resource Id of the deployment script. Example: /subscriptions/{subId}/resourceGroups/{rgName}/providers/Microsoft.Resources/deploymentScripts/{deploymentScriptName}")]
        [ValidateNotNullOrEmpty]
        public string ResourceId { get; set; }
```

Reply:
Same with the other comment - We're consistent with the Deployment(e.g: GetAzureSubscriptionDeployment) cmdlets with this usage. Let me know if you have a strong opinion on this. 

---
In reply to: [390159975](https://github.com/Azure/azure-powershell/pull/11266#discussion_r390159975) [](ancestors = 390159975)

Reply:
I'm OK with that!